react-backend/client/src/components/authentication/newPassword.js

```diff
@@ -8,11 +8,12 @@ import errorMessages from '../errors';
 export default class NewPassword extends Component {
   constructor(props) {
     super(props);
+    const token = this.props.match.params.token;
```

Comment:
IMO it's safer to move this to `componentWillMount` instead of the `constructor`. As far as I understand, the `constructor` will be called only _once_ for the whole lifetime of the app, whereas `componentWillMount` will be called each time the component is mounted onto the DOM.